patches/minecraft/net/minecraft/world/gen/feature/jigsaw/JigsawPattern.java.patch

```diff
@@ -0,0 +1,35 @@
+--- a/net/minecraft/world/gen/feature/jigsaw/JigsawPattern.java
++++ b/net/minecraft/world/gen/feature/jigsaw/JigsawPattern.java
+@@ -21,7 +21,7 @@
+    public static final JigsawPattern field_214949_a = new JigsawPattern(new ResourceLocation("empty"), new ResourceLocation("empty"), ImmutableList.of(), JigsawPattern.PlacementBehaviour.RIGID);
+    public static final JigsawPattern field_214950_b = new JigsawPattern(new ResourceLocation("invalid"), new ResourceLocation("invalid"), ImmutableList.of(), JigsawPattern.PlacementBehaviour.RIGID);
+    private final ResourceLocation field_214951_c;
+-   private final ImmutableList<Pair<JigsawPiece, Integer>> field_214952_d;
++   private final List<Pair<JigsawPiece, Integer>> field_214952_d;
+    private final List<JigsawPiece> field_214953_e;
+    private final ResourceLocation field_214954_f;
+    private final JigsawPattern.PlacementBehaviour field_214955_g;
+@@ -29,7 +29,7 @@
+ 
+    public JigsawPattern(ResourceLocation p_i51397_1_, ResourceLocation p_i51397_2_, List<Pair<JigsawPiece, Integer>> p_i51397_3_, JigsawPattern.PlacementBehaviour p_i51397_4_) {
+       this.field_214951_c = p_i51397_1_;
+-      this.field_214952_d = ImmutableList.copyOf(p_i51397_3_);
++      this.field_214952_d = Lists.newArrayList(p_i51397_3_);
+       this.field_214953_e = Lists.newArrayList();
+ 
+       for(Pair<JigsawPiece, Integer> pair : p_i51397_3_) {
+@@ -56,6 +56,14 @@
+       return this.field_214954_f;
+    }
+ 
++   public void addBuilding(Pair<JigsawPiece,Integer> piece)
```

Comment:
Maybe rather construct the pair internally for the list and pass the values as separate parameters